### ISSUE TITLE
All files uploaded via Auto Upload are modifyed - SHA256 is different from the original

### ISSUE THREAD

**Issue by GAS85** [nextcloud/android#12991]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

1. Setup auto upload
2. Upload photos, Videos
3. Go to the Original Files folder and execute `find -type f -exec sha256sum '{}' \; > ~/2024.hash.phone` to calculate original hash sums.
4. Upload this file to the Server
5. Go to the server folder with uploaded files via `ssh` and run `sha256sum -c 2023.hash.phone` to check SHA 256 sums. Add `| grep OK | wc -l` to the command to see amount of OK files.
6. Be shocked.
```
sha256sum: WARNING: 303 computed checksums did NOT match
10 # This is amount of files that checksums are matched
```
11. Upload file via WebUI and see that SHA256 is different from what was uploaded via Android App, but **it is matching original file SHA256**. On the Left File from Cloud, on the Right fresh uploaded via WebUI: ![Снимок экрана от 2024-05-07 18-55-27](https://github.com/nextcloud/android/assets/6813635/361bed71-bfbb-443e-ae87-d4a8715ff93d)
12. Goto [Memories](https://github.com/pulsejet/memories/) and check EXIF. In original File I have a Location saved, in uploaded via UI it was erased. On the Right File from Cloud, on the Left fresh uploaded via WebUI: ![Снимок экрана от 2024-05-07 18-59-56](https://github.com/nextcloud/android/assets/6813635/cd086486-4a4a-4ea0-89a0-6b6114de80ff)

:cry: I lost tons of GPS locations in Photos that I do not have in Original files anymore.

### Expected behaviour

Files are **never modified** as it was before December 2023. I do check Maps app and have no locations saved since December! 
I have auto update on in google Play, so seems around [3.27.0](https://github.com/nextcloud/android/releases/tag/stable-3.27.0) version was broken.

### Actual behaviour

Never expected that file will be modified in any matter, but seems related to #8622 and #8646

### Android version

14

### Device brand and model

Samsung Galaxy S21 FE

### Stock or custom OS?

Stock

### Nextcloud android app version

3.29.0

### Nextcloud server version

27.1.9

### Using a reverse proxy?

No

### Server error logs

```bash
No related error
```

### Additional information

JPEG and MP4 files are affected

**Comment by TippyTurtle** [comment]
To "fix" (er work around) this on a Pixel 8 Pro running Android 14 do the following:
Settings-->Apps-->Special App access (at the bottom)-->All Files Access-->Nextcloud-->Turn on "Allow access to manage all files"
...in addition to the standard "App Permissions" being set correctly.

I have verified on my phone the sha256sum are the same:
Pixel 8-->Nextcloud App-->InstantUpload-->NextCloud 28.0.5-->Nextcloud Ubuntu File Sync-->sha256sum on local file system
...vs...
Pixel 8-->MTP USB 3 connection-->sha256sum over the MTP mount

Android is "clean" your private data before  it gives it to any app...unless the app has "Special App access" which is not in the normal "App Permissions"

**Comment by joshtrichards** [comment]
What do the ten files that matched have in common?

Were they created after the permission change?

Put another way, now that access is granted, if you take a *new* photo does it also differ or is it a match?

**Comment by TippyTurtle** [comment]
> What do the ten files that matched have in common?
> 
> Were they created after the permission change?
> 
> Put another way, now that access is granted, if you take a _new_ photo does it also differ or is it a match?

(After deleting the copy of all the files on my NextCloud Server) On my phone, if I reconfigure the "Auto upload" to  "Also upload existing files" and it goes to town, the newly uploaded, old photo's, have their exif data (it wasn't stripped this time).  In other words, the files on the phone were not touched by anything, including the Android Nextcloud App.

My Guess:  the normal (many versions) old Android API call to get files will now stripping all the files of meta data before the file is given to the calling app...probably even MS Word Doc's...unless that Android app has "Special App access (at the bottom)-->All Files Access"

**Comment by joshtrichards** [comment]
> My Guess: the normal (many versions) old Android API call to get files will now stripping all the files of meta data before the file is given to the calling app...probably even MS Word Doc's...unless that Android app has "Special App access (at the bottom)-->All Files Access"

No need for guessing. That's expected behavior. Location data for media files is not provided to the Nextcloud Android client unless that permission is granted interactively by the user. That's how Android OS works (these days):

https://developer.android.com/training/data-storage/shared/media#media-location-permission

> To "fix" (er work around) this on a Pixel 8 Pro running Android 14 do the following:
Settings-->Apps-->Special App access (at the bottom)-->All Files Access-->Nextcloud-->Turn on "Allow access to manage all files"

This isn't a workaround. This is precisely why the client asks for those permissions from the user at installation time. They also have the option of choosing the lesser one (which doesn't provide location data to Nextcloud).

In any case, I'm interested in the *reporter's* response rather than guessing what the situation is with those ten files. :-)

**Comment by TippyTurtle** [comment]
> > My Guess: the normal (many versions) old Android API call to get files will now stripping all the files of meta data before the file is given to the calling app...probably even MS Word Doc's...unless that Android app has "Special App access (at the bottom)-->All Files Access"
> 
> No need for guessing. That's expected behavior. Location data for media files is not provided to the Nextcloud Android client unless that permission is granted interactively by the user. That's how Android OS works (these days):
> 
> https://developer.android.com/training/data-storage/shared/media#media-location-permission
> 
> > To "fix" (er work around) this on a Pixel 8 Pro running Android 14 do the following:
> > Settings-->Apps-->Special App access (at the bottom)-->All Files Access-->Nextcloud-->Turn on "Allow access to manage all files"
> 
> This isn't a workaround. This is precisely why the client asks for those permissions from the user at installation time. They also have the option of choosing the lesser one (which doesn't provide location data to Nextcloud).
> 
> In any case, I'm interested in the _reporter's_ response rather than guessing what the situation is with those ten files. :-)

Agreed. It's expected behavior, but what's unexpected is that the app doesn't request these permissions during install. It does ask for the normal permissions. 

It feels like a heavy lift for a normal user because the permissions required are not under the normal app permissions. 

![Screenshot_20240516-133423](https://github.com/nextcloud/android/assets/1476397/c7ae1d93-7251-47c2-94d1-a9f82e7fe33f)

They're under special permissions would you have to go digging for in your system settings after you install the app.

![Screenshot_20240516-133049](https://github.com/nextcloud/android/assets/1476397/23cafb0c-5355-4cc6-a490-34f97a6ea4e5)

**Comment by TippyTurtle** [comment]
> As you see, I grant full access and even before to open a ticket I cancel it, granted again and upload few files via app. Somehow it is not working for me. May be Samsung update caused this, because before December everything was fine.

Yes those are the "normal" permissions that that app requests when you install it.

You need to **additionally** turn on:

Settings-->Apps-->Special App access (at the bottom)-->All Files Access-->Nextcloud-->Turn on "Allow access to manage all files"

Although on Samsung the navigation is probably different.  When you get to the right setting it will look similar to:
![Screenshot_20240516-133049](https://github.com/nextcloud/android/assets/1476397/40ce2c00-0da6-41ed-9416-46a5c3642344)

**Comment by joshtrichards** [comment]
> Agreed. It's expected behavior, but what's unexpected is that the app doesn't request these permissions during install. It does ask for the normal permissions.
> 
> It feels like a heavy lift for a normal user because the permissions required are not under the normal app permissions.

I would agree it would be heavy lifting, but we do request it at install time. :-)

I just uninstalled and reinstalled:

![Screenshot_20240516-172952.png](https://github.com/nextcloud/android/assets/1731941/af606b26-1a3f-40c4-93b4-578bd35cc2dd)

Selecting the right most option brings up the expected screen:

![Screenshot_20240516-173336.png](https://github.com/nextcloud/android/assets/1731941/b5aa3397-e7e1-43ea-8be1-572cd2a7eee4)

Which can either be turned on or not:

![Screenshot_20240516-173338.png](https://github.com/nextcloud/android/assets/1731941/5e329039-c818-47c5-a297-9fae9ee00dba)

**Comment by joshtrichards** [comment]
@GAS85 Don't want to lose your thread:

What is the nature of the ten files that passed the SHA checks? Were they newer media than all the others? Did the originals already lack location data? Etc. I'm trying to figure out why some of your media were okay. It might help determine why your other media are not okay.

**Comment by TippyTurtle** [comment]
> > Agreed. It's expected behavior, but what's unexpected is that the app doesn't request these permissions during install. It does ask for the normal permissions.
> > It feels like a heavy lift for a normal user because the permissions required are not under the normal app permissions.
> 
> I would agree it would be heavy lifting, but we do request it at install time. :-)
> 
> I just uninstalled and reinstalled:
> 
> ![Screenshot_20240516-172952.png](https://private-user-images.githubusercontent.com/1731941/331391033-af606b26-1a3f-40c4-93b4-578bd35cc2dd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU4OTY3NzUsIm5iZiI6MTcxNTg5NjQ3NSwicGF0aCI6Ii8xNzMxOTQxLzMzMTM5MTAzMy1hZjYwNmIyNi0xYTNmLTQwYzQtOTNiNC01NzhiZDM1Y2MyZGQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MTZUMjE1NDM1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTM1MTc1MmZkODRlOWY3ZWY5MzNmMGIwMzc2YWIxZDEyNGY4M2UyMjEwMWFjNzVlMjA1Yzg3MTk4N2MyMjkzMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.cpZJUTtP6oE4cS3SiI3fekEp2sqsAdS4x4fcNkp9kMo)
> 
> Selecting the right most option brings up the expected screen:
> 
> ![Screenshot_20240516-173336.png](https://private-user-images.githubusercontent.com/1731941/331392235-b5aa3397-e7e1-43ea-8be1-572cd2a7eee4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU4OTY3NzUsIm5iZiI6MTcxNTg5NjQ3NSwicGF0aCI6Ii8xNzMxOTQxLzMzMTM5MjIzNS1iNWFhMzM5Ny1lN2UxLTQzZWEtOGJlMS01NzJjZDJhN2VlZTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MTZUMjE1NDM1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MmM0MWY0ZGE3ZWNjYTY4ZjNlMzcwN2VjMDEwM2U2OTdmNWVkNzFhMDFhNDgwNzMzMTcwNDk2NmRmNDcxMGY1MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.LtR56PGELKCpWm3Q7i0icKGcmW5jvW_hrDPncvDjY9c)
> 
> Which can either be turned on or not:
> 
> ![Screenshot_20240516-173338.png](https://private-user-images.githubusercontent.com/1731941/331392318-5e329039-c818-47c5-a297-9fae9ee00dba.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU4OTY3NzUsIm5iZiI6MTcxNTg5NjQ3NSwicGF0aCI6Ii8xNzMxOTQxLzMzMTM5MjMxOC01ZTMyOTAzOS1jODE4LTQ3YzUtYTI5Ny05ZmFlOWVlMDBkYmEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MTZUMjE1NDM1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDcwY2ExNzdlZTZmMDg4NTJmYWI2YmYxM2IyODE5Zjk4ZmViMjNkODljZTM2MTBmN2M1M2E5OTVhNmFlODJjYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.uBOiQBfHXJJOtWVd3vcV6IMLNMCzgC9WiJyW6WWeVkY)

You're exactly right. I hadn't reinstalled in a long time....I guess there is an odd corner case when this happens.  Probably not something the app can fix.

For whatever reason I ended up in this state...and had to hunt down this odd additional setting, when an uninstall/reinstall would have solved it.

Thanks for the clarification.

**Comment by GAS85** [comment]
> @GAS85 Don't want to lose your thread:
> 
> What is the nature of the ten files that passed the SHA checks? Were they newer media than all the others? Did the originals already lack location data? Etc. I'm trying to figure out why some of your media were okay. It might help determine why your other media are not okay.

I double check and they enter do not have location tag, or were uploaded via web UI.

**Comment by GAS85** [comment]
> You need to **additionally** turn on:
> 
> Settings-->Apps-->Special App access (at the bottom)-->All Files Access-->Nextcloud-->Turn on "Allow access to manage all files"

Wow, that was not easy to find. Seem now files uploaded with the same sha, so no modifications to exif.

Totally agree that this must be highlighted to user. I didn't get this prompt after install the app.

**Comment by QuentinLemCode** [comment]
Hello
I encountered the same issue on S23 Ultra, with the app downloaded from Play store. 
When I launched the app, it asked for the permission to access media but not full access. 
The special permission to allow access to all files was unavailable (even in the "special access" settings : nextcloud was not in the list).
I uninstalled the app and installed the one from the github release page and I had the proper permission pop-up asking for all files access and then it worked.
